### PR TITLE
Implement Advection and Curvature Models

### PR DESCRIPTION
**Pull request by Besler**

### CODE REVIEW

**Comment by Besler** [comment]
Example forward/backward advection

advect_negative
advect_positive

**Comment by Besler** [comment]
There will be a small issue in the implementation where boundary conditions are not handled correctly at the edge away from motion. This is because the correct boundary condition is that |\nabla \phi| is 1, but we are doing a mirror boundary condition making it zero.

This can be correct by reinitialization (or other means). It is non-trivial to fix.

**Comment by Besler** [comment]
Here is the curvature example.

curvature

**Comment by Besler** [comment]
@tdkemp Advection and curvature is done! Let me know if you have any thoughts.

Next I will implement the "curvature driven" paper.

File: examples/constant_advection.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+composite.add_layer(ReorderAxes())
+composite.sink = SaveImage(os.path.join(tiff_dir, "Temp"), ".tiff")
+solver.add_sink(composite)
+solver.add_sink(SaveImage(os.path.join(image_dir, "Temp_LS"), ".nii"))
+
+print('Setting up grid')
+rod = Rod(size, spacing, origin, center, radius, longitudinal_vector)()
+solver.grid = DistanceTransform()(rod)
+
+print('Settin up advection velocity field')
```
Comment (tdkemp) [inline examples/constant_advection.py]: Small typo in this line:

"Setting up advection velocity field"

File: wolff/models/mean_curvature_model.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    def constant(self):
+        return self._constant
+
+    @constant.setter
+    def constant(self, value):
+        self._constant = value
+
+    def process(self, grid):
+        # Compute mean curvature
+        curvature = MeanCurvature(multiply_gradient_magnitude=True)(grid)
```
Comment (tdkemp) [inline wolff/models/mean_curvature_model.py]: This is a bit weird, not sure if we can get rid of the double brackets.

**Comment by tdkemp** [comment]
@Besler Looks great! I left a few minor comments.

File: examples/constant_advection.py (hunk 1 above)
Comment (Besler) [reply examples/constant_advection.py]: Good catch!

File: wolff/models/mean_curvature_model.py (hunk 2 above)
Comment (Besler) [reply wolff/models/mean_curvature_model.py]: This is all over the code and stems from the design. Not pretty, but we kind of have to live with it for now.

**Comment by Besler** [comment]
> @Besler Looks great! I left a few minor comments.

Typo fixed, all yours to merge.

**Comment by Besler** [comment]
I will go ahead and merge here so I can develop the next section.

**Pull request merged**